swagger.yaml

```diff
@@ -244,6 +245,101 @@ paths:
           description: Wrong content type (should be JSON)
         422:
           description: Missing paramaters
+  /projects/{projectId}/resources:
+    parameters:
+      - in: path
+        name: projectId
+        schema:
+          type: integer
+        required: true
+    get:
+      tags:
+        - resources
+        - projects
+      summary: Gets metadata of resources in project
+      security:
+        - user_id: []
+      description: Returns a list of metadata of all the resources in the given project
+      responses:
+        200:
+          description: Resource created successfully
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/ResourceMetadata"
+        400:
+          description: Malformed Request
+        404:
+          description: Project not found
+        422:
+          description: Missing paramaters
+    post:
+      tags:
+        - resources
+        - projects
+      summary: Creates new resource in project
+      security:
+        - user_id: []
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/ResourceMetadata"
+      description: Creates a new resource with given name for the given project
+      responses:
+        200:
+          description: Resource created successfully
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Resource"
+        400:
+          description: Malformed Request
+        404:
+          description: Project not found
+        409:
+          description: Duplicate resource
+        415:
+          description: Wrong content type (should be JSON)
+        422:
+          description: Missing paramaters
+  /projects/{projectId}/resources/{resourceId}:
+    parameters:
+      - in: path
+        name: projectId
+        schema:
+          type: integer
+        required: true
+      - in: path
+        name: resourceId
+        schema:
+          type: integer
+        required: true
```

Comment:
Do we need endpoint for updating resource? I'd start with simply adding and getting resource

Reply:
Didn't see your comment, it's a bit late to go back now, but without the PUT endpoint the GET doesn't make sense that much

Reply:
Why not? You would have POST to upload, DELETE to delete and GET to download

Reply:
Oh, I haven't spotted that POST only create id for resource and PUT actually uploads it. Why this way? Seems a bit odd to me

Reply:
I made it like this for consistency, since we send the resource name in the body (as in all other POSTs) so sending the file data would require a multipart form request